File: src/test/java/org/revcloud/vader/matchers/AnyMatchersTest.java
package org.revcloud.vader.matchers;

import static org.hamcrest.Matchers.lessThan;
import static org.revcloud.vader.matchers.IntMatchers.inRangeInclusive;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class AnyMatchersTest {

  @Test
  void anyOfOrNullMatchersTest() {
    Assertions.assertTrue(AnyMatchers.anyOfOrNull(inRangeInclusive(1, 31)).matches(null));
    Assertions.assertTrue(AnyMatchers.anyOfOrNull(inRangeInclusive(1, 31)).matches(1));
    Assertions.assertFalse(AnyMatchers.anyOfOrNull(inRangeInclusive(1, 31)).matches(0));
  }

  @Test
  void anyOfMatchersTest() {
    Assertions.assertTrue(AnyMatchers.anyOf(inRangeInclusive(1, 31), lessThan(10)).matches(0));
  }
}

File: src/test/kotlin/org/revcloud/vader/matchers/AnyMatchersTest.kt
package org.revcloud.vader.matchers

import io.kotest.core.datatest.forAll
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe
import org.hamcrest.Matchers

class AnyMatchersTest : FunSpec({

    test("anyOf") {
        anyOf(inRangeInclusive(1, 31), Matchers.nullValue()).matches(null) shouldBe true
        forAll(1 to 31) {
            anyOf(inRangeInclusive(1, 31), Matchers.nullValue()).matches(null) shouldBe true
        }
    }
})
